src/components/AddPuzzleForm.tsx

```diff
@@ -29,45 +33,64 @@ const AddPuzzleForm = () => {
   const [priceHasError, setPriceHasError] = useState(false);
   const [pieceCountHasError, setPieceCountHasError] = useState(false);
 
+  const [isSuccessful, setIsSuccessful] = useState(false);
   const [openSuccessMessage, setOpenSuccessMessage] = useState(false);
+  const [image, setImage] = useState('');
+  const [message, setMessage] = useState('');
 
   const handleSubmit = (event: MouseEvent) => {
     event.preventDefault();
     checkIfErrors();
-    if(category !== '' && missingPieceCount !== '' && price !== '' && quality !== '' && pieceCount !== '') {
-      const newPuzzle: ICleanedPuzzleObject = {
-        id: Date.now().toString(),
-        category: category,
-        missingPieces: missingPieceCount,
-        pieceCount: pieceCount,
-        quality: quality,
-        availability: true,
-        price: price,
-        image: 'https://img.buzzfeed.com/buzzfeed-static/static/2020-04/28/14/asset/1593bcadc012/sub-buzz-488-1588084568-26.jpg'
-      }
+    if(category  && missingPieceCount  && price  && quality  && pieceCount && image) {
```

Comment:
I think it was a good idea to separate all of this out into their own functions!